### PR TITLE
chore: clean up uploaded cse files for AgentBaker Windows E2E automatically

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Clean up uploaded cse files (cse logs and cse packages) for AgentBaker Windows E2E automatically.
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: e2e/windows/cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+set -euxo pipefail
+
+# delete blobs created 3 days ago
+EXPIRATION_IN_HOURS=72
+(( expirationInSecs = ${EXPIRATION_IN_HOURS} * 60 * 60 ))
+(( deadline=$(date +%s)-${expirationInSecs%.*} ))
+dateOfdeadline=$(date -d @${deadline} +"%Y-%m-%dT%H:%M:%S+00:00")
+
+# two containers need to be cleaned up now
+CONTAINER_LIST=("cselogs" "csepackages")
```
Comment (AbelHu) [inline e2e/windows/cleanup.sh:12]: could we support deleting the containers created by @wanqingfu's PR?

File: e2e/windows/cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+for CONTAINER_NAME in "${CONTAINER_LIST[@]}"
+do 
+    result=$(az storage blob list -c $CONTAINER_NAME --account-name $STORAGE_ACCOUNT_NAME --account-key $MAPPED_ACCOUNT_KEY -o json \
+    | jq -r --arg time "$dateOfdeadline" '.[] | select(.properties.creationTime < $time)' \
+    | jq -r '.name')
+
+    for item in $result
+    do
+        az storage blob delete -c $CONTAINER_NAME --account-name $STORAGE_ACCOUNT_NAME --account-key $MAPPED_ACCOUNT_KEY -n $item
```
Comment (AbelHu) [inline e2e/windows/cleanup.sh:22]: Add no-wait?

File: .pipelines/e2e-windows.yaml [hunk 3]
```diff
@@ -70,3 +70,16 @@ stages:
       windowsDistro: aks-windows-2022-containerd-gen2
       windowsNodeImageVersion: AKSWindows-2022-containerd-gen2-test-2023.02.07
       storageAccount: akswinstore2022gen2
+
+- stage: backfill_cleanup_outdated_resources
```
Comment (AbelHu) [inline .pipelines/e2e-windows.yaml:74]: how do you plan to run this stage? Is there any issue if it is run by multiple PRs?

File: e2e/windows/cleanup.sh (hunk 1 above)
Comment (junjiezhang1997) [reply e2e/windows/cleanup.sh:12]: Not sure why we need to delete the containers. We use one container for one kubernetes version. Every time the resource group is deleted automatically, we will also clean up the files in the container before we upload new files from new created resource group.

File: e2e/windows/cleanup.sh (hunk 2 above)
Comment (junjiezhang1997) [reply e2e/windows/cleanup.sh:22]: It seems there's no-wait parameter for this command. And in my test, this command executes quickly and is not a long-running operation.

File: .pipelines/e2e-windows.yaml (hunk 3 above)
Comment (junjiezhang1997) [reply .pipelines/e2e-windows.yaml:74]: I missed the scenario for multiple pipelines to delete outdated resources at the same time. I think I can run this script in the pipeline of creating new resources, which could avoid conflict.

File: e2e/windows/cleanup.sh (hunk 1 above)
Comment (AbelHu) [reply e2e/windows/cleanup.sh]: I think that we can add this after Qingfu's PR is merged.

**Review by AbelHu — APPROVED** [review]
LGTM

**Pull request merged**